### PR TITLE
Fix landing page

### PR DESCRIPTION
**Pull request by arnaud-morvan**

### CODE REVIEW

File: docker/qgisserver/geomapfish_qgisserver/accesscontrol.py [hunk 1]
```diff
@@ -226,19 +226,24 @@ def get_ogcserver_accesscontrol_config(self) -> None:
...
                 raise GMFException("The map parameter should not be provided")
             return self.ogcserver_accesscontrol
-        if "MAP" not in parameters:
-            raise GMFException("The map parameter should be provided")
-        if parameters["MAP"] not in self.ogcserver_accesscontrols:
+
+        if "MAP" in parameters:
+            config_uri = parameters["MAP"]
+        else:
+            config_uri = self.serverInterface().configFilePath()
```
Comment (sbrunner) [inline docker/qgisserver/geomapfish_qgisserver/accesscontrol.py:238]: What does `self.serverInterface().configFilePath()`?
Especially when we on the landing page?

File: geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py [hunk 2]
```diff
@@ -174,7 +174,7 @@ def _proxy_callback(
     ) -> Response:
         if self.request.matched_route.name.endswith("_path"):
             url = url.clone()
-            url.path = self.request.path
+            url.path = "/".join(self.request.matchdict["path"])
```
Comment (sbrunner) [inline geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py:177]: For me, this is suspect, is the `QGIS_SERVER_LANDING_PAGE_PREFIX` configured correctly? 
To `mapserv_proxy/qgis`

File: docker/qgisserver/geomapfish_qgisserver/accesscontrol.py (hunk 1 above)
Comment (arnaud-morvan) [reply docker/qgisserver/geomapfish_qgisserver/accesscontrol.py:238]: When we are on the landing page, browser retrieve index.json and try GetMap with https://localhost:8484/mapservproxy/qgis/project/<hash>
And it that case we do not have any MAP parameter, but `self.serverInterface().configFilePath()` contains the project path.

File: geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py (hunk 2 above)
Comment (arnaud-morvan) [reply geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py:177]: Yes, but we still should request http://qgisserver:8080/index.html not http://qgisserver:8080/mapserv_proxy/qgis/index.html
Maybe I miss something, but it works like that.

**Comment by arnaud-morvan** [comment]
@sbrunner : I could propose to make a short meet to see the current state together.

File: geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py (hunk 2 above)
Comment (arnaud-morvan) [reply geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py]: I setted the value of QGIS_SERVER_LANDING_PAGE_PREFIX in demo.
But it did not changed anything, I do not really understand if we should use QGIS_SERVER_LANDING_PAGE_PREFIX env var or X-Qgis-Service-Url.

I see in doc that:
QGIS_SERVER_LANDING_PAGE_PREFIX is Landing page base URL path prefix
It is used here: https://github.com/qgis/QGIS/blob/7ae1b318731fda9a98f29bb8d795a274d7ab0124/src/server/services/landingpage/qgslandingpagehandlers.cpp#L77

But I see this about X-Qgis-Service-Url:
https://github.com/qgis/QGIS/blob/9563b040ef37aee51db54a8074227ca4e2886767/src/server/qgsserverprojectutils.h#L441:L461

In all case API still return http://qgisserver:8080

File: doc/integrator/backend_qgis.rst [hunk 3]
```diff
@@ -224,6 +224,6 @@ To have the landing page you should:
 - Define the ``QGIS_SERVER_LANDING_PAGE_PROJECTS_DIRECTORIES`` or the
   ``QGIS_SERVER_LANDING_PAGE_PROJECTS_PG_CONNECTIONS`` environment variable.
 - Create a new OGC server with a name e.g. ``qgis`` and no ``MAP`` in the query string.
-- Define the ``QGIS_SERVER_LANDING_PAGE_PREFIX`` to, in our example, ``/mapserv_proxy/qgis``.
```
Comment (sbrunner) [inline doc/integrator/backend_qgis.rst:227]: Why removing this?

File: geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py (hunk 2 above)
Comment (arnaud-morvan) [reply geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py]: In fact it does not work anymore when settings QGIS_SERVER_LANDING_PAGE_PREFIX

Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/views/mapserverproxy.py]: I do some test with QGIS server alone (with our image) and As I understand we should have `QGIS_SERVER_LANDING_PAGE_PREFIX` and here we should have the pull path...

**Comment by arnaud-morvan** [comment]
I have error when preparing for tests locally:
```
docker-compose exec -T tests alembic --config=/opt/c2cgeoportal/commons/alembic.ini --name=main \
	upgrade head
Traceback (most recent call last):
  File "/usr/local/bin/alembic", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/alembic/config.py", line 588, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/usr/local/lib/python3.8/dist-packages/alembic/config.py", line 582, in main
    self.run_cmd(cfg, options)
  File "/usr/local/lib/python3.8/dist-packages/alembic/config.py", line 559, in run_cmd
    fn(
  File "/usr/local/lib/python3.8/dist-packages/alembic/command.py", line 300, in upgrade
    script = ScriptDirectory.from_config(config)
  File "/usr/local/lib/python3.8/dist-packages/alembic/script/base.py", line 205, in from_config
    util.coerce_resource_to_filename(script_location),
  File "/usr/local/lib/python3.8/dist-packages/alembic/util/pyfiles.py", line 58, in coerce_resource_to_filename
    ref = compat.importlib_resources.files(tokens[0])
  File "/usr/local/lib/python3.8/dist-packages/importlib_resources/_common.py", line 22, in files
    return from_package(get_package(package))
  File "/usr/local/lib/python3.8/dist-packages/importlib_resources/_common.py", line 53, in get_package
    resolved = resolve(package)
  File "/usr/local/lib/python3.8/dist-packages/importlib_resources/_common.py", line 44, in resolve
    return cand if isinstance(cand, types.ModuleType) else importlib.import_module(cand)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'c2cgeoportal_commons'
```
I do not understand as I do not get error when installing package c2cgeoportal_common.
Tried to install package and import it manually and get same error.

**Comment by sbrunner** [comment]
I'm working on that :-)

**Review by sbrunner — APPROVED** [review]

**Comment by sbrunner** [comment]
Thanks :-)

**Pull request merged**